test/unit/markup/base_test.py

```diff
@@ -19,3 +23,11 @@ def test_apply_xslt_stylesheets(self):
         assert 'xslt-' in self.markup.apply_xslt_stylesheets(
             '../data/example_config.xml'
         )
+
+    @patch('kiwi.markup.base.etree.parse')
+    def test_apply_xslt_stylesheets_nonXML(self, mock_parse):
+        mock_parse.side_effect = etree.XMLSyntaxError('not-XML', '<', 1, 1)
+        with raises(KiwiConfigFileFormatNotSupported):
+            self.markup.apply_xslt_stylesheets(
+                '../data/example_config.xml'
+            )
```

Comment:
I think the test could confuse people when reading. It sets a raise side effect on etree.parse which is good but it calls apply_xslt_stylesheets() with an existing and XML correct file :) As you set the side effect on etree.parse it's actually immaterial which config file is passed to the method. Therefore I would write this as:

```
with raises(KiwiConfigFileFormatNotSupported):
    self.markup.apply_xslt_stylesheets('artificial_and_invalid_XML_markup')
```

What do you think ?

Reply:
Works for me ;-) I had been trying to leverage the iterable aspect for side_effect to get it to do the right thing in the original code workflow, where the etree.parse() is called first for the xslt and then for the provided descript, and then changed the code workflow to do the relevant etree.parse(description) as the first thing it does to make it easier to write the unit tests ;-)